packages/core/test/oracle/OptimisticOracle.js

```diff
@@ -154,6 +154,72 @@ contract("OptimisticOracle", function(accounts) {
+      await optimisticRequester.requestPrice(identifier, requestTime, "0x", collateral.address, "0");
+    });
+    it("Should return false when no price was ever proposed", async function() {
+      const result = await optimisticOracle.hasPrice(optimisticRequester.address, identifier, requestTime, "0x");
+      assert.equal(result, false);
+    });
+    it("Should return false when price is proposed but not past liveness", async function() {
+      await collateral.approve(optimisticOracle.address, totalDefaultBond, { from: proposer });
+      await optimisticOracle.proposePrice(optimisticRequester.address, identifier, requestTime, "0x", correctPrice, {
+        from: proposer
+      });
+      const result = await optimisticOracle.hasPrice(optimisticRequester.address, identifier, requestTime, "0x");
+      assert.equal(result, false);
+    });
+    it("Should return true when price is proposed and past liveness but not settled", async function() {
+      await collateral.approve(optimisticOracle.address, totalDefaultBond, { from: proposer });
+      await optimisticOracle.proposePrice(optimisticRequester.address, identifier, requestTime, "0x", correctPrice, {
+        from: proposer
+      });
+      await timer.setCurrentTime((await timer.getCurrentTime()).add(await optimisticOracle.defaultLiveness()));
+      const result = await optimisticOracle.hasPrice(optimisticRequester.address, identifier, requestTime, "0x");
+      assert.equal(result, true);
+    });
+    it("Should return false when price is proposed and disputed", async function() {
+      await collateral.approve(optimisticOracle.address, totalDefaultBond, { from: proposer });
+      await optimisticOracle.proposePrice(optimisticRequester.address, identifier, requestTime, "0x", correctPrice, {
+        from: proposer
+      });
+
+      await collateral.approve(optimisticOracle.address, totalDefaultBond, { from: disputer });
+      await optimisticOracle.disputePrice(optimisticRequester.address, identifier, requestTime, "0x", {
+        from: disputer
+      });
+
+      const result = await optimisticOracle.hasPrice(optimisticRequester.address, identifier, requestTime, "0x");
+      assert.equal(result, false);
+    });
+    it("Should return true when price is proposed and past liveness but not settled", async function() {
+      await collateral.approve(optimisticOracle.address, totalDefaultBond, { from: proposer });
+      await optimisticOracle.proposePrice(optimisticRequester.address, identifier, requestTime, "0x", correctPrice, {
+        from: proposer
+      });
+      await timer.setCurrentTime((await timer.getCurrentTime()).add(await optimisticOracle.defaultLiveness()));
+      const result = await optimisticOracle.hasPrice(optimisticRequester.address, identifier, requestTime, "0x");
+      assert.equal(result, true);
+    });
+    it("Should return true when price is proposed, disputed and resolved by dvm", async function() {
```

Comment:
Do you think you should also test the path where the price is proposed, past liveness, and settled?